src/libs/merge/kdbmerge.c

```diff
@@ -952,6 +953,17 @@ KeySet * elektraMerge (KeySet * our, Key * ourRoot, KeySet * their, Key * theirR
 		       int strategy, Key * informationKey)
 {
 	ELEKTRA_LOG ("cmerge starts with strategy %d (see kdbmerge.h)", strategy);
+	if (strategy == MERGE_STRATEGY_EQUAL)
+	{
+		if ((strcmp (keyName (ourRoot), keyName (theirRoot)) == 0) && (strcmp (keyName (theirRoot), keyName (baseRoot)) == 0))
```

Comment:
With this check you do not know if the keysets are the same?

Reply:
#1306 is the reason why I started this (= strategy unchanged). I thought if we merge user/test/a with user/test/a and user/test/a, how could there be any difference?

I took a look at the mount code and don't really have an idea how to include the merge yet. Consequently, this strategy was a bit guessed, too.

I couldn't find any comparison function for key sets. Did I overlook this or does it really not exist?

Reply:
> I thought if we merge user/test/a with user/test/a and user/test/a, how could there be any difference?

Ok, then different question: why does it depend on the strategy if we anyway merge "the same thing".

> I couldn't find any comparison function for key sets. Did I overlook this or does it really not exist?

It does not exist as there are so many variants of what it could mean that two KeySets are equal. And it is quite trivial to implement. One variant is implemented in compare_keyset tests/cframework/tests.h

I think in your case you want a quite strict variant: i.e. also the metadata needs to be the same.